### ISSUE TITLE
delete reference when unsharing a resource

### ISSUE THREAD

**Issue by butonic** [cs3org/reva#1275]
how are we going to clean up a reference when the owner unshares a folder? 

AFAICT the gateway usershareprolider will get a RemoveShare request.
1. we delete the share in the usershareprovider,
2. and we can delete the grant in the storageprovider of the owner.

But we also need to delete the reference of the recipient. how do we do that?

**Comment by butonic** [comment]
we could make the gateway delete references if it encounters invalid ones?

**Comment by ishank011** [comment]
Yes, that could be one option. Or we could run a background cron. We need a periodic garbage collection functionality for a lot of cases - public shares, invalid references, expired tokens in OCM. So it'd be good to have a package for this.
